src/Types/PlutusData.purs

```diff
@@ -23,3 +36,40 @@ derive instance Generic PlutusData _
 
 instance Show PlutusData where
   show x = genericShow x
+
+-- On-chain PlutusData. Do we need this representation to separate when writing
+-- the Contract Monad?
+newtype BuiltinPlutusData = BuiltinPlutusData PlutusData
```

Comment:
Do we really need to maintain a distinction here like Plutus does?

Reply:
I would be very glad to see the back of this. My only concern is, do we want a distinction when writing the contract Monad? In particular, do we need to separate on-chain and off chain data so we can't off-chain functions on chain and vice versa? I feel like I don't have a good understanding of the bigger picture ATM (I guess with regards to data serialisation/deserialisation, Ogmios and related helpers).

If so, we still need to keep `ToPlutusData` right? Although have it map to `PlutusData` instead of `BuiltinPlutusData`?

Reply:
Yes, we should still have the two typeclasses. For `BuiltinData`, the issue with Plutus is that it is quite possible to maximize code reuse between on-chain and off-chain Plutus code. Our `Contract` interface will be in PS, though. So I don't think we need to worry about the distinction